src/bin/pg_autoctl/monitor.c

```diff
@@ -1214,22 +1215,29 @@ monitor_set_formation_number_sync_standbys(Monitor *monitor, char *formation,
  * on the monitor.
  */
 bool
-monitor_remove_by_hostname(Monitor *monitor, char *host, int port)
+monitor_remove_by_hostname(Monitor *monitor, char *host, int port, bool force)
 {
 	SingleValueResultContext context = { { 0 }, PGSQL_RESULT_BOOL, false };
 	PGSQL *pgsql = &monitor->pgsql;
-	const char *sql = "SELECT pgautofailover.remove_node($1, $2)";
-	int paramCount = 2;
-	Oid paramTypes[2] = { TEXTOID, INT4OID };
-	const char *paramValues[2];
+	const char *sql = "SELECT pgautofailover.remove_node($1, $2, $3)";
+	int paramCount = 3;
+	Oid paramTypes[3] = { TEXTOID, INT4OID, BOOLOID };
+	const char *paramValues[3];
 
 	paramValues[0] = host;
 	paramValues[1] = intToString(port).strValue;
+	paramValues[2] = force ? "true" : "false";
 
 	if (!pgsql_execute_with_params(pgsql, sql,
 								   paramCount, paramTypes, paramValues,
 								   &context, &parseSingleValueResult))
 	{
+		/* if we fail to find the node we want to remove, we're good */
+		if (strcmp(context.sqlstate, STR_ERRCODE_UNDEFINED_OBJECT) == 0)
```

Comment:
Could this specific error code happen for other reasons? e.g. the function was removed during an upgrade.

Reply:
I believe that would raise a `undefined_function` error (see https://www.postgresql.org/docs/current/errcodes-appendix.html) rather than a very generic `undefined_object` one. I think we're good here.